docs/blockchain-integrations-framework/index.md

```diff
@@ -0,0 +1,229 @@
+
+Integration testing uses the `bif integration soak` command and a TOML file. The TOML file simplifies managing the many variables involved.
+
+The TOML file for the Blockchain Integrations Framework has two settings sections:
+
+- Network settings: Variables that configure which chain you use for testing and the private key for the wallet you want to use for deploying contracts
+- Test settings: Node count and test duration settings to control the behavior of the test
+
+**Example TOML file**
+
+```toml
+# network settings
+[networks]
+
+    [networks.evm]
+
+        [networks.evm.base]
+        name = "EVM"
+        private_keys = ["WALLET_PRIVATE_KEY"]
+        ws_urls = []
+        http_urls = []
+        chain_id = 1337
+        simulated = false
+        chainlink_transaction_limit = 5000
+        transaction_timeout = "2m"
+        minimum_confirmations = 1
+        gas_estimation_buffer = 1000
+
+        [networks.evm.overrides.goerli]
+        name = "Goerli"
+        ws_urls = ["wss://eth-goerli.g.alchemy.com/v2/<ALCHEMY_KEY>"]
+        chain_id = 5
+
+        [networks.evm.overrides.optimism_goerli]
+        name = "Optimism Goerli"
+        ws_urls = ["wss://opt-goerli.g.alchemy.com/v2/<ALCHEMY_KEY>"]
+        chain_id = 420
+
+        [networks.evm.overrides.palm]
+        name = "Palm"
+        ws_urls = ["wss://palm"]
+        chain_id = 11297108099
+
+# test settings
+[tests]
+
+    [tests.v1-7-0]
+
+        [tests.v1-7-0.base]
+        keep_environments="Never" # Always | OnFail | Never
+        # Image repo to pull the Chainlink image from
+        chainlink_image="public.ecr.aws/chainlink/chainlink"
+        # Version of the Chainlink image to pull
+        chainlink_version="1.7.0-nonroot"
+        # Name of the person running the tests (no spaces)
+        chainlink_env_user="My-Name" 
+        test_log_level="info" # info | debug | trace
+        node_count=6
+        test_duration=15 # minutes
+        contract_count=2
+        node_funding=0.1 # ETH
+        round_timeout=15 # minutes
+        expected_round_time=2 # minutes
+        time_between_rounds=1 # minutes
+
+        [tests.v1-7-0.overrides.shorter-test]
+        test_duration=2 # minutes
+
+    [tests.v1-9-0]
+
+        [tests.v1-9-0.base]
+        keep_environments="Never" # Always | OnFail | Never
+        # Image repo to pull the Chainlink image from
+        chainlink_image="public.ecr.aws/chainlink/chainlink"
+        # Version of the Chainlink image to pull
+        chainlink_version="1.9.0"
+        # Name of the person running the tests (no spaces)
+        chainlink_env_user="My-Name"
+        test_log_level="info" # info | debug | trace
+        node_count=6  
+        test_duration=15 # minutes
+        contract_count=2
+        node_funding=0.1 # ETH
+        round_timeout=15 # minutes
+        expected_round_time=2 # minutes
+        time_between_rounds=1 # minutes
+
+        [tests.v1-9-0.overrides.shorter-test]
+        test_duration=2 # minutes
```

Comment:
More up to date TOML, this one contains a `nodes` section that is used by the infra module:
```toml
# example TOML config

# node settings
[nodes]

    [nodes.goerli]

        [nodes.goerli.base]
        eth_url = "wss://goerli"
        eth_http_url = "https://goerli"
        eth_chain_id = "5"

# network settings
[networks]

    [networks.evm]

        [networks.evm.base]
        name = "EVM"
        evm_keys = ["YOUR_EVM_KEY_HERE"]
        ws_urls = []
        chain_id = 1337
        simulated = false
        chainlink_transaction_limit = 5000
        transaction_timeout = "2m"
        minimum_confirmations = 1
        gas_estimation_buffer = 1000
        client_implementation = "Ethereum"

        [networks.evm.overrides.goerli]
        name = "Goerli"
        ws_urls = ["wss://goerli"]
        chain_id = 5
        transaction_timeout = "5m"
        minimum_confirmations = 5

        [networks.evm.overrides.palm]
        name = "Palm"
        ws_urls = ["wss://palm"]
        chain_id = 11297108099

        [networks.evm.overrides.rinkeby]
        name = "Rinkeby"
        ws_urls = ["wss://rinkeby"]
        chain_id = 4

        [networks.evm.overrides.optimism]
        name = "Optimism"
        ws_urls = ["wss://optimism"]
        chain_id = 420
        client_implementation = "Optimism"

# test settings
[tests]

    [tests.v1-7-0]

        [tests.v1-7-0.base]
        keep_environments="Never" # Always | OnFail | Never
        chainlink_image="public.ecr.aws/chainlink/chainlink" # Image repo to pull the Chainlink image from
        chainlink_version="1.7.0-nonroot" # Version of the Chainlink image to pull
        chainlink_env_user="Satoshi-Nakamoto" # Name of the person running the tests (change to your own)
        test_log_level="info" # info | debug | trace
        test_duration=15 # minutes
        contract_count=2
        node_funding=0.1 # ETH
        round_timeout=15 # minutes
        expected_round_time=2 # minutes
        time_between_rounds=1 # minutes

        [tests.v1-7-0.overrides.shorter-test]
        test_duration=2 # minutes
```
You can add a line about the `nodes` section as well, saying that the infra module uses it to configure environment variables on nodes that it deploys.